arctic/store/_pickle_store.py

```diff
@@ -53,13 +58,14 @@ def write(self, arctic_lib, version, symbol, item, previous_version):
         # Pickle, chunk and store the data
         collection = arctic_lib.get_top_level_collection()
         # Try to pickle it. This is best effort
-        version['blob'] = _MAGIC_CHUNKED
-        pickled = lz4.compressHC(cPickle.dumps(item, protocol=cPickle.HIGHEST_PROTOCOL))
+        version['blob'] = _MAGIC_CHUNKEDV2
+        pickled = cPickle.dumps(item, protocol=cPickle.HIGHEST_PROTOCOL)
 
         for i in xrange(int(len(pickled) / _CHUNK_SIZE + 1)):
-            segment = {'data': Binary(pickled[i * _CHUNK_SIZE : (i + 1) * _CHUNK_SIZE])}
-            sha = checksum(symbol, segment)
+            data = compress(pickled[i * _CHUNK_SIZE: (i + 1) * _CHUNK_SIZE])
```

Comment:
This runs the compression serially - it would be better to call `compress_array` with all the chunks, which will run the compression in parallel if the Cython library is available.